cobang: add support for aarch64-linux

dvaerum opened the pull request:
## Description of changes

Added `"aarch64-linux"` to `meta.platforms`

## Things done

I have built cobang on the architecture `aarch64-linux` and it works for me, so can we add support for the `aarch64-linux` architecture?

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

superherointj commented on pkgs/applications/misc/cobang/default.nix:
```diff
@@ -99,6 +99,6 @@ buildPythonApplication rec {
     maintainers = [ ];
     mainProgram = "cobang";
-    platforms = [ "x86_64-linux" ];
+    platforms = [ "x86_64-linux" "aarch64-linux" ];
```
```suggestion
    platforms = lib.platforms.linux;
```

Aleksanaa replied:
Yeah, just assume it works, since it's python

dvaerum replied:
I have updated the pull request to use `lib.platform.linux` instead

superherointj replied:
> `lib.platform.linux`

![image](https://github.com/user-attachments/assets/617a73ae-f5b2-40ab-b492-50b5c3bf47c7)

dvaerum replied:
Thanks @superherointj, for noticing that I forgot a `s` in `platform` to should have been fixed

Aleksanaa commented on pkgs/applications/misc/cobang/default.nix:
```diff
@@ -99,6 +99,6 @@ buildPythonApplication rec {
     license = licenses.gpl3Only;
     maintainers = [ ];
```
Would you mind adding yourself to maintainers? Or you can add me to maintainers

dvaerum replied:
I will try to add myself (and you) as maintainers, I just need to get myself added to `maintainers/maintainer-list.nix`.
I have made a https://github.com/NixOS/nixpkgs/pull/331326 which I am waiting on

superherointj replied:
Cherry-pick that commit here. And respect order of commits.

dvaerum replied:
@superherointj I am trying my best to respect the project, I read the document about how to add oneself as a maintainer, but there is a lot left out, like what the subject of the pull request should be. Maybe it was written somewhere else, but when I did not find it.

superherointj replied:
> @superherointj I am trying my best to respect the project, I read the document about how to add oneself as a maintainer, but there is a lot left out, like what the subject of the pull request should be. Maybe it was written somewhere else, but when I did not find it.

**Familiarity**. You cannot guess right what you don't know. But yes, documentation could be better, but realize, different people are at different level at different matters, so there isn't a documentation that fits everyone needs. Some know git or GitHub well, others don't... Most conventions are straightforward. If you want to change something, open a PR, reviewers will debate it [but is a tiring process].

In truth, everything sucks, proposing code sucks, reviewing sucks, committing sucks even more. (Having to own shit sucks.) This is unpaid work, right?

I ask myself everyday why I'm still doing this. Idk. I honestly don't know. =]

If a problem arise, we'll figure out a solution. Don't worry.

dvaerum replied:
@superherointj thanks for the link (I will read it in more details and may have follow-up questions), I think that there probably should be a link to the conventions in the document
https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
Because it was there I reading about how to add myself as an maintainer.

superherointj replied:
I agree that documentation could be improved.

dvaerum replied:
@Aleksanaa 
Okay, I should have fixed the order of the commits (and the commit text there I added myself to the maintainer list)

@superherointj 
I hope everything looks good now

dvaerum committed:
cobang: set `platform` to support `lib.platforms.linux`

I have built cobang on the architecture `aarch64-linux` and it works for me.
Since the program is written in python I will assume that it will work on all Linux architectures so I have set `platform` to support `lib.platforms.linux`

superherointj commented:
Also [always] squash commits. (It should be a single commit for this change.)

superherointj commented:
ping @dvaerum

superherointj commented:
Please only add clean commits. By clean I mean, no:

> Merge branch 'NixOS:master' into patch-1

First commit should be:
> maintainers: add dvaerum

Order matters because of bisecting.

Second commit should be:
> cobang: add support aarch64-linux
> cobang: add dvaerum as maintainer

You could make the `cobang` change a single commit. But then, you have to add to the body of commit message what you are doing. You are doing two actions. So you comment that.

Options are: 3 commits or 2 commits (adding information to body of commit). Up to you.

dvaerum committed:
maintainers: add dvaerum

superherointj commented on pkgs/applications/misc/cobang/default.nix:
```diff
@@ -97,8 +97,8 @@ buildPythonApplication rec {
     homepage = "https://github.com/hongquan/CoBang";
     license = licenses.gpl3Only;
-    maintainers = [ ];
+    maintainers = with maintainers; [ dvaerum aleksana ];
```
```suggestion
    maintainers = with maintainers; [ dvaerum aleksana ];
```
I'd default to alphabetical order. While the list may be small now, it could grow in the future.

superherointj commented:
On commit message:

Things I'd do differently:
- Avoid special characters such as ``. We never know how this could affect something.
- Keep comment objective and concise. For example:
  - cobang: add aarch64-linux platform
  - cobang: add dvaerum as maintainer

This is not stopping this PR from being merged. I'm waiting CI atm.

superherointj commented:
![image](https://github.com/user-attachments/assets/cb2c3045-23f7-48d7-a599-0172c7fc8331)

Is this package building for you?

superherointj commented:
#330241

dvaerum commented:
> ![image](https://private-user-images.githubusercontent.com/5861043/354412384-cb2c3045-23f7-48d7-a599-0172c7fc8331.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1NTUyOTcsIm5iZiI6MTcyMjU1NDk5NywicGF0aCI6Ii81ODYxMDQzLzM1NDQxMjM4NC1jYjJjMzA0NS0yM2Y3LTQ4ZDctYTU5OS0wMTcyYzdmYzgzMzEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDFUMjMyOTU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTc1MDRlZjUxYWQwYTA2MTM0ZDI3OWQ5YWVkYTQwMGM1Zjk0N2MwNDRkNTM3OGFmNTIxNWQxY2U1NzQ3OGQxNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.gOdxEoWV4UN8x-EsdyEB9vxGjOOJ__14Z36_hNJVL8s)
> 
> Is this package building for you?

No, not after the 6 of July I think it was, it seems to be a problem with a unit test which are triggered, I have an `overriteAttrs` which disables `nativeCheckInputs` so that the build will work.
```nix
python-zbar_p = pkgs.python312Packages.python-zbar.overrideAttrs (oldAttrs: rec {
  nativeCheckInputs = [ ];
});
```
I have not figured out yet which unittest there needs to be disabled for the build to work, but `cobang` works on my PC.
It was something I wanted to look into it more after this merge request and try to make a pull request with a fix and maybe become a maintainer of that package too, since it does not have one currently

superherointj commented:
> I have not figured out yet which unittest there needs to be disabled for the build to work, 

I just gave you the solution.

> but `cobang` works on my PC.

The reference must be nixpkgs master. And it is failing at nixpkgs master. While I understand the problem was not introduced by you, and the changes you are proposing are unrelated. We should be fixing it. And I already gave you the solution.

> It was something I wanted to look into it more after this merge request (made it before I learned that the `python-zbar` package broke) and try to make a pull request with a fix and maybe become a maintainer of that package too, since it does not have one currently

Let's do this in this PR since I'm already reviewing this. (You are saving yourself time.)

dvaerum committed:
python312Packages.python-zbar: fix build

dvaerum commented:
> To fix it, I suggest you remove from:
> 
> > pkgs/development/python-modules/python-zbar/default.nix
> 
> The lines:
> 
> ```
>   # distutils usage in setup.py
>   disabled = pythonAtLeast "3.12";
> ```
> 
> Then, it builds fine.
> 
> This also means an extra commit:
> 
>     * `python312Packages.python-zbar: fix build`

I hope I understood your instructions correctly 😇

dvaerum commented:
> I hope I understood your instructions correctly 😇

Looks like I misunderstood something or there is another problem :confused:

superherointj commented:
Remove `pythonAtLeast` reference since it is no longer used.

superherointj commented:
Result of `nixpkgs-review pr 326368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>cobang</li>
    <li>cobang.dist</li>
    <li>python312Packages.python-zbar</li>
    <li>python312Packages.python-zbar.dist</li>
  </ul>
</details>

superherointj commented:
ping @dvaerum

dvaerum commented:
Hey @superherointj, thanks for the pointers

superherointj commented:
Result of `nixpkgs-review pr 326368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python312Packages.python-zbar</li>
    <li>python312Packages.python-zbar.dist</li>
  </ul>
</details>

superherointj commented:
Result of `nixpkgs-review pr 326368` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>cobang</li>
    <li>cobang.dist</li>
    <li>python312Packages.python-zbar</li>
    <li>python312Packages.python-zbar.dist</li>
  </ul>
</details>
